sqlglot/optimizer/qualify_columns.py

```diff
@@ -60,8 +60,8 @@ def qualify_columns(
         _pop_table_column_aliases(scope.derived_tables)
         using_column_tables = _expand_using(scope, resolver)
 
-        if schema.empty and expand_alias_refs:
-            _expand_alias_refs(scope, resolver)
+        if (schema.empty or dialect.FORCE_EARLY_ALIAS_REF_EXPANSION) and expand_alias_refs:
+            _expand_alias_refs(scope, resolver, expand_only_groupby=(dialect == "bigquery"))
```

Comment:
Hello @VaggelisD 

Should we provide the `expand_only_groupby` via qualify_columns parameters? (we can infer it with your logic if user provide None).  

The context is if I extend the Bigquery dialect to MyBigquery dialect, this change will not work anymore, so I want to force it using a parameter passed to the function caller instead.

Reply:
Can't you just set the flag in your dialect?

Reply:
Sure, `expand_only_groupby` inside the dialect is a good idea.

Reply:
Ah you were talking about the expand only group logic, I completely missed it sorry. We could probably parameterize it, cc @VaggelisD

Reply:
https://github.com/tobymao/sqlglot/pull/3702